package.json

```diff
@@ -15,9 +15,9 @@
   "scripts": {
     "dev": "webpack --node-env development --progress",
     "watch": "webpack --node-env development --progress --watch",
-    "watch:module": "LIBRARY_TARGET=module webpack --node-env development --progress --watch",
+    "watch:module": "NODE_ENV=production vite build --watch",
```

Comment:
```suggestion
    "watch:module": "vite build --mode production --watch",
```

Reply:
BTW here it does work as we want to build in production to prevent that vue2 issue, but generally it does not work, as `--mode development` does not set the `NODE_ENV` (it sets it to 'production').

Reply:
Do you need the `NODE_ENV` ?

Reply:
Thats how e.g. vue decides to show more verbose warnings

Reply:
Weird, it should also be assigning NODE_ENV
https://github.com/vitejs/vite/blob/2b1ffe86328f9d06ef9528ee117b61889893ddcc/packages/vite/src/node/plugins/define.ts#L27

Reply:
Just tested it and the NODE_ENV is indeed set properly with `--mode`
https://codesandbox.io/p/sandbox/throbbing-sky-x6gpw2

Reply:
This only works when **not** building in library mode:
https://github.com/vitejs/vite/blob/2b1ffe86328f9d06ef9528ee117b61889893ddcc/packages/vite/src/node/plugins/define.ts#L19

That is why I need to add those defines manually in the vite config:
https://github.com/susnux/nextcloud-vite-config/blob/91b2b564c6894daefe1ffe035a3d6fd339054c03/lib/baseConfig.ts#L112

And you can test that quite easy: Create a basic vite config, put `console.log(process.env)` on top of the config and run `vite --mode development build`:
```
% npx vite -c a.js --mode development build
mode is: production
vite v4.3.9 building for development...
✓ 1 modules transformed.
dist/test.mjs  0.06 kB │ gzip: 0.08 kB
✓ built in 53ms
```

```
% cat a.js
console.log("mode is:", process.env.NODE_ENV)
export default {
        build: {
                lib: {
                        entry: 'b.js',
                        formats: ['es'],
                },
        },
}
```

Reply:
> This only works when **not** building in library mode: [vitejs/vite@`2b1ffe8`/packages/vite/src/node/plugins/define.ts#L19](https://github.com/vitejs/vite/blob/2b1ffe86328f9d06ef9528ee117b61889893ddcc/packages/vite/src/node/plugins/define.ts#L19)

I love it ahaha, This is such a specific issue! Why do we always end up in the rabbit hole here :sob: :rofl:

Reply:
Done with nextcloud-vite-config beta 13.